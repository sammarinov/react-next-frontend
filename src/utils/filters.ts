import { IFeatured, IArticle } from "@Redux/app/type";

export const filterFeatured = (featured: Array<IFeatured>, limit: number = -1) => {
    featured.sort((a, b) => {
        if (a.topFeatured && !b.topFeatured) return 1;
        if (!a.topFeatured && b.topFeatured) return -1;
        return parseInt(a.price) - parseInt(b.price);
    })

    if (limit > 0) return featured.slice(0, limit);
    return featured;
}

export const filterArticles = (articles: Array<IArticle>, limit: number = -1) => {
    articles.sort((a, b) => {
        return Date.parse(a.created) - Date.parse(b.created);
    })

    if (limit > 0) return articles.slice(0, limit);
    return articles;
}