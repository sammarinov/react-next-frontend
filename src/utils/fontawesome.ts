import { library } from '@fortawesome/fontawesome-svg-core';

import {
    faArrowRight
} from '@fortawesome/free-solid-svg-icons';

import {
    faTwitter,
    faFacebook,
    faInstagram,
    faYoutube
} from '@fortawesome/free-brands-svg-icons';

library.add(
    faArrowRight,
    faTwitter,
    faFacebook,
    faInstagram,
    faYoutube
)