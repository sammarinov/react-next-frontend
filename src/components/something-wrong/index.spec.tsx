import * as React from 'react';

import { render } from '@Test/utils';
import SomethingWrong from './index';

describe('SomethingWrong', () => {
  it('should match snapshot', () => {
    const { container } = render(<SomethingWrong />);
    expect(container).toMatchSnapshot();
  });
  it('should render without fail', () => {
    const { getByText } = render(<SomethingWrong />);
    expect(getByText('SOMETHING_WRONG')).toBeTruthy();
    expect(getByText('GO_BACK')).toBeTruthy();
  });
});
