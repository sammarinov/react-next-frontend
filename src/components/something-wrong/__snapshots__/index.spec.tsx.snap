// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SomethingWrong should match snapshot 1`] = `
.c2 {
  width: 100%;
  height: 100%;
  min-width: 1.25rem;
  min-height: 1.25rem;
  border: 1px solid transparent;
  outline: none;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.3125rem;
  grid-template-columns: 1fr;
  cursor: pointer;
  border-radius: 0.3125rem;
  background-color: #05a8aa;
  border-color: transparent;
  color: #ffffff;
}

.c2:focus {
  outline: none;
}

.c0 {
  font-weight: 800;
  color: #000;
  text-align: center;
  font-size: 2.5em;
  padding-top: 20px;
}

.c1 {
  height: 40px;
  width: 50%;
  margin: 0 auto 20px;
}

@media screen and (max-width:400px) {
  .c0 {
    padding-left: 20px;
    padding-right: 20px;
    font-size: 2em;
  }
}

<div>
  <h1
    class="c0"
  >
    SOMETHING_WRONG
  </h1>
  <div
    class="c1"
  >
    <button
      class="c2"
    >
      GO_BACK
    </button>
  </div>
</div>
`;
